File: src/16.ts
import fs from 'fs'
import { range } from './helper/iterable-helper'

const source = fs.readFileSync('./res/popular-names.txt', 'utf-8')
const solve16 = (size: number): Array<string> => {
    const rows = source.split('\n')
    const blockCount = Math.ceil(rows.length / size)
    return range(blockCount)
        .map(i => rows
            .slice(size * i, size * (i + 1))
            .join('\n')
        )
}
console.log(
    solve16(10)
)

File: src/16.py
import math
from typing import List

from .helper.file_helper import read_string

source = read_string("./res/popular-names.txt")


def solve16(result_len: int) -> List[str]:
    rows = source.split("\n")
    rows_len = len(rows)
    part_size = math.ceil(rows_len / result_len)
    return list("\n".join(rows[start : (start + part_size)]) for start in range(0, rows_len, part_size))


print(solve16(100)[0])
print()
print(solve16(100)[-1])
